### ISSUE TITLE
darshan does not produce error when DARSHAN_LOGPATH_BY_ENV doesn't point to valid logpath

### ISSUE THREAD

**Issue by shanedsnyder** [darshan-hpc/darshan#222]
In GitLab by @shanedsnyder on Jan 23, 2017, 17:06

When using a Darshan install configured using the `DARSHAN_LOGPATH_BY_ENV` environment variable (instead of configuring Darshan to use a static directory for log files), Darshan does not indicate any error when the log path is set incorrectly.

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Jan 30, 2017, 21:26

I can't reproduce this.

Here's my darshan-runtime configure line:

`../configure --with-mem-align=8 --with-log-path-by-env=DARSHAN_LOG_DIR --with-jobid-env=NONE --prefix=/home/shane/software/darshan/darshan-dev/install/ CC=mpicc CFLAGS='-Wall -g -O0'`

When running a test program with `DARSHAN_LOG_DIR` set to a bad path, I get:

`darshan library warning: unable to open log file /tmp/test/loges/shane_ior_id18376_1-30-76110-3858722356085012778.darshan_partial`

@glennklockwood, do you have any more details about what was causing the issue? The behavior should be to:

1.) Check if DARSHAN_LOGFILE env variable is set, and if so, store the file directly at the name it points to.

2.) Check all of the environment variables specified at configure time (using `--with-log-path-by-env`), and use the first variable that actually points to something. The environment variable should point to the directory to store the logs in, and the directory does **not** have to be in the traditional darshan format (year/month/day). 

3.) Use the log directory specified at configure time using `--with-log-path` switch. This directory **does** have to be structured in the traditional darshan log format (using the `darshan-mk-log-dirs.pl` script).

4.) An error should be generated if it can't be determined where to store the log file in 1-3.

**Comment by shanedsnyder** [comment]
In GitLab by @glennklockwood on Jan 31, 2017, 13:38

One notable difference is that I am compiling with `--enable-mmap-logs`.  If this doesn't seem to be the culprit, I can run more rigorous tests to narrow down where this problem arises.

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Feb 9, 2017, 14:16

I confirmed this should be working properly using the mmap code path as well. If the user accidentally types in an invalid path (either in the argument to `--with-log-path` configure switch or in the value of environment variables which were arguments to the `--with-log-path-by-env` configure switch), Darshan should output an error whenever it tries to open the file since the path is bad. 

Darshan was **not** outputting a warning if it never determined a log path to try to store log files at (i.e., `--with-log-path` was not specified and no log paths were found in environment variables provided to `--with-log-path-by-env`). I changed this in commit c41966aa so that now we at least get a warning in stderr that Darshan doesn't know where to store its log.

I think this issue should be covered, so I'll close for now, but let me know if you have more details or still notice a problem some times. Maybe something else weird is happening.
